microsoft/testsuites/gpu/gpusuite.py

```diff
@@ -0,0 +1,104 @@
+
+
+@TestSuiteMetadata(
+    area="gpu",
+    category="functional",
+    description="""
+    This test suite runs the gpu test cases.
+    """,
+)
+class gpu(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            This test case verfies kernel support for gpu.
+            TODO: validate the test params against GPU support test matrix
+                    when test matrix is available.
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu],
+        ),
+    )
+    def validate_gpu_support(self, case_name: str, node: Node) -> None:
+        gpu_feature = node.features[Gpu]
+        gpu_feature.check_support()
+
+    @TestCaseMetadata(
+        description="""
+            This test case verfies if gpu drivers are loaded fine.
+
+            Steps:
+            1. Validate if the VM SKU is supported for GPU.
+            2. Install LIS drivers if not already installed for Fedora and its
+                derived distros. Reboot the node
+            3. Install required gpu drivers on the VM and reboot the node. Validate gpu
+                drivers can be loaded successfully.
+
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu, SerialConsole],
+        ),
+    )
+    def validate_load_driver(self, case_name: str, node: Node) -> None:
+        gpu_feature = node.features[Gpu]
+        case_path = self._create_case_log_path(case_name)
+
+        gpu_vendor = gpu_feature.install_compute_sdk()
+        self.log.info(
+            f"{gpu_vendor} sdk is installed. Will reboot to validate driver is loaded."
+        )
+
+        reboot_tool = node.tools[Reboot]
+        reboot_tool.reboot_and_check_panic(case_path)
+
+        # TODO - More logic to be added if support for more vendors added.
+        lsmod_tool = node.tools[Lsmod]
+        if not lsmod_tool.module_exists(gpu_vendor):
+            raise LisaException(
+                f"{gpu_vendor} GPU driver is not loaded after VM restart!"
+            )
+
+    @TestCaseMetadata(
+        description="""
+            This test case verifies the gpu adapter count.
+
+            Steps:
+            1. Find the expected gpu count for the node.
+            2. Find the actual gpu count using lsvmbus output.
+            3. Compare expected and actual value. Fail if the values don't match.
+
+        """,
+        requirement=simple_requirement(
+            supported_features=[Gpu, SerialConsole],
+        ),
+    )
+    def validate_gpu_adapter_count(self, case_name: str, node: Node) -> None:
+        expected_device_count = node.capability.gpu_count
+        # gpu_name = "Tesla"
+
+        lsvmbus_tool = node.tools[Lsvmbus]
+        device_list = lsvmbus_tool.get_device_channels_from_lsvmbus()
+        actual_device_count = 0
+        gpu_bridge_count = 0
+        for device in device_list:
+            # Device Id for  gpu SKUs
+            if "47505500" in device.device_id:
```

Comment:
I'd put those alien numbers in variables (constants). Also, use GPU instead of gpu mentions as much as possible. Finally, fix typos :)